azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java

```diff
@@ -84,6 +84,28 @@
     this.dependencyTransferManager = dependencyTransferManager;
   }
 
+  /**
+   * Constructor
+   * This constructor is used for Containerized execution which does not need projectCache
+   *
+   * @param projectStorageManager projectStorageManager
+   * @param dependencyTransferManager dependencyStorageManager
+   */
+  public FlowPreparer(final ProjectStorageManager projectStorageManager,
```

Comment:
Better to use inheritance over here. Parent can be without cache. Child can be with Cache. From Flow container, we can use object of parent.

Reply:
Can do it in follow up PR?

Reply:
It will be good to have a clean code. 
If you want to take it in next PR then please create ticket for it.

Reply:
Aaah nevermind, I confused myself. No the cache is not used. Please notice the new constructor of FlowPreparer.

Reply:
Can you please create a ticket to have parent and child classes for FlowPreparer? child will have caching mechanism.